crates/transcode/src/lib.rs

```diff
@@ -273,6 +279,40 @@ impl ContractMessageTranscoder {
     fn messages(&self) -> impl Iterator<Item = &MessageSpec<PortableForm>> {
         self.metadata.spec().messages().iter()
     }
+    /// Compares the contract's environment type with a provided type definition.
+    pub fn compare_type(
```

Comment:
This method belongs better together with the rest in `env_check` rather than in the transcoder. We could just add a getter to access `self.metadata`.